### PR TITLE
java: maven http request timeout added

### PR DESCRIPTION
**Pull request by kripanshdevtron**
maven package search http request timeout made configurable.

### CODE REVIEW

File: java/packagescanner.go [hunk 1]
```diff
@@ -34,12 +34,14 @@ var (
...
 const DefaultSearchAPI = `https://search.maven.org/solrsearch/select`
+const DefaultRequestTimeout = 10
 
 // ScannerConfig is the struct used to configure a Scanner.
 type ScannerConfig struct {
 	// API is a URL endpoint to a maven-like REST API.
 	// The default is DefaultSearchAPI.
-	API string `yaml:"api" json:"api"`
+	API               string `yaml:"api" json:"api"`
+	APIRequestTimeout int    `yaml:"api_request_timeout" json:"api_request_timeout"`
```
Comment (hdonnay) [inline java/packagescanner.go:44]:
```suggestion
	APIRequestTimeout time.Duration `yaml:"api_request_timeout" json:"api_request_timeout"`
```

The timeout should just be a duration.

File: java/packagescanner.go [hunk 2]
```diff
@@ -34,12 +34,14 @@ var (
 )
 
 const DefaultSearchAPI = `https://search.maven.org/solrsearch/select`
+const DefaultRequestTimeout = 10
```
Comment (hdonnay) [inline java/packagescanner.go:37]: If this is configurable, I'd like the default timeout to be much smaller, like 2 seconds.

**Review by hdonnay — CHANGES_REQUESTED** [review]
Please squash the commits and have them follow the project conventions.

File: java/packagescanner.go (hunk 1 above)
Comment (kripanshdevtron) [reply java/packagescanner.go:44]: done this to prevent extra memory, time.Duration contains extra params, which are not required.

Comment (kripanshdevtron) [reply java/packagescanner.go:44]: I have done this to prevent add extra memory to struct.

Comment (hdonnay) [reply java/packagescanner.go:44]: That's not a concern for a low-usage struct like this, compared to the documentation and language-level benefit. A time.Duration is an int64 under the hood, so there's no memory saved by making this an int.

**Comment by hdonnay** [comment]
The code changes look good -- please squash the commits down to one and add a signoff.

**Pull request closed without merging**